src/stackdriver-nozzle/stackdriver/metrics_buffer.go

```diff
@@ -16,12 +16,83 @@
 
 package stackdriver
 
-import "reflect"
+import (
+	"context"
+	"github.com/mitchellh/hashstructure"
+	"reflect"
+	"sync"
+	"time"
+)
 
 type MetricsBuffer interface {
 	PostMetric(*Metric)
```

Comment:
maybe we want a `Flush`/`IsEmpty` to support [comment](https://github.com/cloudfoundry-community/stackdriver-tools/pull/76/commits/521c7641df5c3c767ca994b860752668a832ed3b#diff-d23e533d7fb5c6f3666b3d9f396c7085R53)